om4labs/diags/moc/moc.py

```diff
@@ -0,0 +1,306 @@
+    x = np.array(ds_gridspec.x.to_masked_array())[::2, ::2]
+    y = np.array(ds_gridspec.y.to_masked_array())[::2, ::2]
+
+    # depth coordinate
+    depth = ds_topog.deptho.to_masked_array()
+    depth = np.where(np.isnan(depth), 0.0, depth)
+    z = get_z(ds, depth, varname)
+
+    # basin code
+    basin_code = ds_basin.basin.to_masked_array()
+
+    # vmo
+    vmo = ds[varname].to_masked_array()
+
+    return x, y, z, basin_code, vmo
+
+
+def calculate(vmo, basin_code):
+    """Main computational script"""
+
+    print("Calculating")
+    msftyyz = compute_msftyyz(vmo, basin_code)
+
+    return msftyyz
+
+
+def plot(y, z, msftyyz, label=None):
+    """Plotting script"""
+
+    def _findExtrema(
+        ax, y, z, psi, min_lat=-90.0, max_lat=90.0, min_depth=0.0, mult=1.0
+    ):
+        """Function to annotate the max/min values on the MOC plot"""
+        psiMax = mult * np.amax(
+            mult * np.ma.array(psi)[(y >= min_lat) & (y <= max_lat) & (z < -min_depth)]
+        )
+        idx = np.argmin(np.abs(psi - psiMax))
+        (j, i) = np.unravel_index(idx, psi.shape)
+        ax.plot(y[j, i], z[j, i], "kx")
+        ax.text(y[j, i], z[j, i], "%.1f" % (psi[j, i]))
+
+    def _plotPsi(y, z, psi, ci, title, cmap=None):
+        """Function to plot zonal mean streamfunction"""
+        plt.contourf(y, z, psi, levels=ci, cmap=cmap, extend="both")
+        cbar = plt.colorbar()
+        plt.contour(y, z, psi, levels=ci, colors="k")
+        plt.gca().set_yscale("splitscale", zval=[0, -2000, -6500])
+        plt.title(title)
+        cbar.set_label("[Sv]")
+        plt.ylabel("Elevation [m]")
+
+    print("Making Plot.")
+    z = z.min(axis=-1)
+    yy = y[1:, :].max(axis=-1) + 0 * z
+
+    psi = msftyyz * 1.0e-9
+
+    ci = m6plot.formatting.pmCI(0.0, 40.0, 5.0)
+    cmap = palettable.cmocean.diverging.Balance_20.get_mpl_colormap()
+
+    fig = plt.figure(figsize=(8.5, 11))
+    ax1 = plt.subplot(3, 1, 1)
+    psiPlot = psi[0, 0]
+    _plotPsi(yy, z, psiPlot, ci, "Atlantic MOC [Sv]", cmap=cmap)
+    _findExtrema(ax1, yy, z, psiPlot, min_lat=26.5, max_lat=27.0)
+    _findExtrema(ax1, yy, z, psiPlot, max_lat=-33.0)
+    _findExtrema(ax1, yy, z, psiPlot)
+
+    ax2 = plt.subplot(3, 1, 2)
+    psiPlot = psi[0, 1]
+    _plotPsi(yy, z, psiPlot, ci, "Indo-Pacific MOC [Sv]", cmap=cmap)
+    _findExtrema(ax2, yy, z, psiPlot, min_depth=2000.0, mult=-1.0)
+    _findExtrema(ax2, yy, z, psiPlot)
+
+    ax3 = plt.subplot(3, 1, 3)
+    psiPlot = psi[0, 2]
+    _plotPsi(yy, z, psiPlot, ci, "Global MOC [Sv]", cmap=cmap)
+    _findExtrema(ax3, yy, z, psiPlot, max_lat=-30.0)
+    _findExtrema(ax3, yy, z, psiPlot, min_lat=25.0)
+    _findExtrema(ax3, yy, z, psiPlot, min_depth=2000.0, mult=-1.0)
+    plt.xlabel(r"Latitude [$\degree$N]")
+
+    plt.subplots_adjust(hspace=0.2)
+
+    if label is not None:
+        plt.suptitle(label)
+
+    return fig
+
+
+def run(args):
+    """Function to call read, calc, and plot in sequence"""
+
+    # parameters
```

Comment:
this become unnecessary if we pass a dict to run instead of the parser namespace

Reply:
addressed in dfd833b